### PR TITLE
reflection probe for custom pipeline

### PR DESCRIPTION
**Pull request by xubing0906**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/rendering/custom/define.ts [hunk 1]
```diff
@@ -496,6 +498,66 @@ export function buildShadowPass (passName: Readonly<string>,
...
+
+    const probePassRTName = `reflectionProbePassColor${cameraName}`;
+    const probePassDSName = `reflectionProbePassDS${cameraName}`;
+
+    probe.updateCameraDir(faceIdx);
+
+    if (!ppl.containsResource(probePassRTName)) {
+        ppl.addRenderTexture(probePassRTName, Format.RGBA8, width, height, renderWindow);
+        ppl.addDepthStencil(probePassDSName, Format.DEPTH_STENCIL, width, height, ResourceResidency.MANAGED);
+        ppl.updateRenderWindow(probePassRTName, renderWindow);
```
Comment (GengineJS) [inline cocos/rendering/custom/define.ts]: 这个得放外面

File: cocos/rendering/custom/define.ts [hunk 2]
```diff
@@ -496,6 +498,66 @@ export function buildShadowPass (passName: Readonly<string>,
...
+        AccessType.WRITE, AttachmentType.DEPTH_STENCIL,
+        getLoadOpOfClearFlag(probeCamera.clearFlag, AttachmentType.DEPTH_STENCIL),
+        StoreOp.STORE,
+        probeCamera.clearFlag,
+        new Color(probeCamera.clearDepth, probeCamera.clearStencil, 0, 0));
+    probePass.addRasterView(probePassRTName, passView);
+    probePass.addRasterView(probePassDSName, passDSView);
+    const passBuilder = probePass.addQueue(QueueHint.RENDER_OPAQUE);
+    passBuilder.addSceneOfCamera(camera, new LightInfo(), SceneFlags.REFLECTION_PROBE);
+    updateCameraUBO(passBuilder as unknown as any, probeCamera, ppl);
```
Comment (GengineJS) [inline cocos/rendering/custom/define.ts:558]: 这个有必要再添加么？addSceneOfCamera得过程就有添加camera ubo吧？

Comment (xubing0906) [reply cocos/rendering/custom/define.ts:558]: addSceneOfCamera确实有添加，但是我这里想用反射探针的虚拟相机去重新更新ubo

File: cocos/rendering/reflection-probe/reflection-probe-stage.ts [hunk 3]
```diff
@@ -96,7 +96,7 @@ export class ReflectionProbeStage extends RenderStage {
     public render (camera: Camera) {
         const pipeline = this._pipeline;
         const cmdBuff = pipeline.commandBuffers[0];
-        this._probeRenderQueue.gatherRenderObjects(this._probe!, camera.scene!);
+        this._probeRenderQueue.gatherRenderObjects(this._probe!, camera.scene!, cmdBuff);
         pipeline.pipelineUBO.updateCameraUBO(this._probe!.camera);
```
Comment (GengineJS) [inline cocos/rendering/reflection-probe/reflection-probe-stage.ts:100]: 这个是为了兼容旧管线么？如果是的话，可以加个判断，判断下cc.rendering是否为undefined

Comment (xubing0906) [reply cocos/rendering/reflection-probe/reflection-probe-stage.ts:100]: 这个传过去是为了下一步去做instance，旧管线也需要

Comment (GengineJS) [reply cocos/rendering/reflection-probe/reflection-probe-stage.ts:100]: 新管线可以不需要，不然你先标记个todo吧，因为现在新管线还有部分沿用旧管线的ubo，后面都要移除的

**Review by GengineJS — APPROVED** [review]

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by star-e — APPROVED** [review]

**Pull request merged**